Homework02/perceptron.py

```diff
@@ -0,0 +1,51 @@
+"""
+perceptron class
+
+Created: 01.11.21, 20:04
+
+Author: LDankert
+"""
+import numpy
+import numpy as np
+
+
+class Perceptron:
+
+    # Constructor
+    def __init__(self, input_units):
+        self.weights = np.random.randn(input_units)
+        self.bias = np.random.randn()
+        self.alpha = 1
+        self.activation = 0
+
+    # Calculates the activation of the perceptron, takes 1d array
+    def forward_step(self, inputs):
+        # raise an exception if the number of inputs not match number of weights
+        if np.size(inputs) != np.size(self.weights):
+            raise Exception ("Inputs number doesn't match weight numbers")
+        else:
+            # multiply the weights with corresponding inputs
+            activation = np.sum((inputs*self.weights))
```

Comment:
Or, we first calculate the net_input, i.e. the sum of inputs & weights multiplied, then add bias - which we could do in one line:
`self.net_input = np.dot(self.input, self.weights) + self.bias`

Then we apply the sigmoid activation function and return it:
`self.activation = self.sigmoid(self.net_input) 
  return self.activation `

I think it's important to distinguish the steps between calculating net_input (i.e., the sum) and activation of the neuron.

Reply:
good hint changed that, but we don't need the return statement, because we already stored the activation inside the perception itself. So we can access it where ever we want